flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/partition/PartitionTableTest.java

```diff
@@ -0,0 +1,107 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.taskexecutor.partition;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Tests for the {@link PartitionTable}.
+ */
+public class PartitionTableTest extends TestLogger {
+
+	private static final JobID JOB_ID = new JobID();
+	private static final ResultPartitionID PARTITION_ID = new ResultPartitionID();
+
+	@Test
+	public void testEmptyTable() {
+		final PartitionTable table = new PartitionTable();
+
+		// an empty table should always return an empty collection
+		Collection<ResultPartitionID> partitionsForNonExistingJob = table.stopTrackingPartitions(new JobID());
+		assertNotNull(partitionsForNonExistingJob);
+		assertEquals(0, partitionsForNonExistingJob.size());
+
+		assertFalse(table.hasTrackedPartitions(new JobID()));
+	}
+
+	@Test
+	public void testStartTrackingPartition() {
+		final PartitionTable table = new PartitionTable();
+
+		table.startTrackingPartition(JOB_ID, PARTITION_ID);
+
+		assertTrue(table.hasTrackedPartitions(JOB_ID));
+	}
+
+	@Test
+	public void testStopTrackingAllPartitions() {
+		final PartitionTable table = new PartitionTable();
+
+		table.startTrackingPartition(JOB_ID, PARTITION_ID);
+
+		Collection<ResultPartitionID> storedPartitions = table.stopTrackingPartitions(JOB_ID);
+		assertThat(storedPartitions, contains(PARTITION_ID));
+		assertFalse(table.hasTrackedPartitions(JOB_ID));
+	}
+
+	@Test
+	public void testStopTrackingSinglePartition() {
+		final ResultPartitionID partitionId2 = new ResultPartitionID();
+		final PartitionTable table = new PartitionTable();
+
+		table.startTrackingPartition(JOB_ID, PARTITION_ID);
+		table.startTrackingPartition(JOB_ID, partitionId2);
+
+		table.stopTrackingPartition(JOB_ID, partitionId2);
+		assertTrue(table.hasTrackedPartitions(JOB_ID));
+
+		Collection<ResultPartitionID> storedPartitions = table.stopTrackingPartitions(JOB_ID);
+		assertThat(storedPartitions, contains(PARTITION_ID));
+		assertFalse(table.hasTrackedPartitions(JOB_ID));
+	}
+
+	@Test
+	public void testStopTrackingSpecificPartitions() {
```

Comment:
`testStopTrackingSinglePartition` and `testStopTrackingSpecificPartitions` could be deduplicated if you like, only one line is different.